spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -239,7 +239,10 @@ def reset_namespace(self, warning=False, message=False):
         reset_str = _("Remove all variables")
         warn_str = _("All user-defined variables will be removed. "
                      "Are you sure you want to proceed?")
-        kernel_env = self.kernel_manager._kernel_spec.env
+        try:
```

Comment:
Please add a comment above this line with the following text:

    # This is necessary to make resetting variables work in external
    # kernels.
    # See spyder-ide/spyder#9505